flink-runtime/src/main/java/org/apache/flink/runtime/source/coordinator/SourceCoordinator.java

```diff
@@ -218,15 +218,12 @@ void announceCombinedWatermark() {
                 operatorName);
 
         for (Integer subtaskId : subTaskIds) {
-            // when subtask have been finished, do not send event.
-            if (!context.hasNoMoreSplits(subtaskId)) {
```

Comment:
Could you explain/refer to what will happen in tasks that don't have splits? I guess it will be just ignored?

What if the task/operator has switched to FINISHED?